core/src/main/java/com/amplifyframework/core/category/Category.java

```diff
@@ -83,7 +83,7 @@ public final synchronized void configure(
                 for (P plugin : getPlugins()) {
                     String pluginKey = plugin.getPluginKey();
                     JSONObject pluginConfig = configuration.getPluginConfig(pluginKey);
-                    plugin.configure(pluginConfig, context);
+                    plugin.configure(pluginConfig != null ? pluginConfig : new JSONObject(), context);
```

Comment:
Is this how we handle this on other platforms? I'd imagine we want users to explicitly configure their use of any plugin via the CLI, so I'm not sure this is ideal. What problem are you tackling here?

Reply:
Idle thought: In general I'm anti-ternary for readability - do we have an opinion in our own team style guide?

Reply:
The goal here is to prevent the issue we're having when the dataStore (or any other category) is not in the `amplifyconfiguration.json` file but the developer adds it as a plugin. Definitely more than one way to skin the cat here...let me give it some more thought

Reply:
I think upgrading to this non-null object ~here is probably a good idea. Jotted down my thinking --

-------
Let's suppose we have a config file like:
```
{
    "fooCategory": {
        "plugins": {
            "fooCategoryPlugin": {
            }
        }
     },
     "barCategory": {
         "plugins": {
             "barCategoryPlugin": {
                "key1": "val1"
             }
         }
     },
     "bazCategory": {
         "plugins": {
         }
     }
}
```
Potentially, the specification of these three categories imply three distinct things:
1. For _foo_, it looks to me like "Hey, use this plugin with default values."
2. For _bar_, it looks like "Hey, use this plugin, and use my values."
3. For _baz_, it looks like: "Huh? What's a baz plugin? we're not expecting to use that kind of a plugin."

If that's so, then we might have three corresponding ways to represent this in Java:
1. `new JSONObject()` with no contents;
2. `new JSONObject().put("key1", "val2")` representing the chosen keys;
3. `null`.

The discrepancy arrises could situations where the config file enumerates config for plugins _p1...pn_, but the user, at runtime, enumerates `π1... πn` via `addPlugin(...)`.

For each plugin, there six code paths to consider, while `configure(...)`ing a plugin at run-time:
1. File has populated config, user doesn't add plugin.
2. File has empty config, user doesn't add plugin.
3. File doesn't have config, user doesn't add plugin.
4. File has populated config, user adds plugin.
5. File has empty config, user adds plugin.
6. File doesn't have config, user adds plugin.

1-3 are probably a non-issue - we can just ignore the unused plugin blocks.
4 - We should probably validate the config block, then use the values
5 - We should whatever default plugin values we can, throw only if config file missing some required value
6 - This is the interesting one, I guess.